src/composables/useTxStatus.ts

```diff
@@ -1,14 +1,69 @@
-const pendingCount = ref(0);
+import { useStorage } from '@vueuse/core';
+import getProvider from '@snapshot-labs/snapshot.js/src/utils/provider';
+import { PendingTransaction } from '@/helpers/interfaces';
+
+const PENDING_TRANSACTIONS_STORAGE_KEY = 'snapshot.pendingTransactions';
+const pendingTransactions = useStorage(
+  PENDING_TRANSACTIONS_STORAGE_KEY,
+  [] as PendingTransaction[]
+);
 
 export function useTxStatus() {
-  watch(pendingCount, () => {
-    if (pendingCount.value < 0) {
-      pendingCount.value = 0;
+  const createPendingTransaction = () => {
+    const createdAt = Date.now();
+    const id = createdAt.toString();
+    const tx = {
+      id,
+      network: '1',
+      createdAt,
+      txId: null
+    };
+    pendingTransactions.value.push(tx);
+    return id;
+  };
+
+  const updatePendingTransaction = (
+    id: string,
+    payload: Partial<PendingTransaction>
+  ) => {
+    const tx = pendingTransactions.value.find(tx => tx.id === id);
+    if (tx) {
+      Object.assign(tx, payload);
     }
-  });
-  return { pendingCount };
-}
+  };
+
+  const removePendingTransaction = (id: string) => {
+    const tx = pendingTransactions.value.find(tx => tx.id === id);
+    if (tx) {
+      pendingTransactions.value = pendingTransactions.value.filter(
+        tx => tx.id !== id
+      );
+    }
+  };
+
+  const restorePendingTransactions = () => {
```

Comment:
1. Use `Array.prototype.forEach()` instead of `Array.prototype.map()` in `restorePendingTransactions()` to better convey your intent and avoid confusion.

2. If many pending transactions exist, `restorePendingTransactions()` may cause rate limiting or performance issues due to numerous requests to the provider. Consider adding a delay between requests to address this issue.